manager/scheduler/scheduler.go

```diff
@@ -418,9 +422,10 @@ func (s *Scheduler) buildNodeHeap(tx store.ReadTx, tasksByNode map[string]map[st
 
 	i := 0
 	for _, n := range nodes {
-		var resources api.Resources
+		var resources *api.Resources
 		if n.Description != nil && n.Description.Resources != nil {
-			resources = *n.Description.Resources
+			resources = &api.Resources{NanoCPUs: n.Description.Resources.NanoCPUs,
+				MemoryBytes: n.Description.Resources.MemoryBytes}
```

Comment:
Maybe this should be `resources = n.Description.Resources.Copy()` so that it doesn't have to copy the fields individually?

Then again, I think your other PR (about generic resources) will change this anyway.